frontend/src/components/Collections/components/Grid/components/CollectionsGrid/index.tsx

```diff
@@ -42,29 +52,44 @@ const CollectionsGrid: FC<Props> = ({
 function renderCollections(
   collections: CollectionResponse[],
   displayVisibility?: VISIBILITY_TYPE,
-  accessType?: ACCESS_TYPE
+  accessType?: ACCESS_TYPE,
+  revisionsEnabled?: boolean
 ) {
   const collectionElements = [] as Array<ReactChild>;
-  sortByCreatedAtDescending(collections).forEach(({ id, visibility }) => {
-    if (!displayVisibility || visibility === displayVisibility) {
-      collectionElements.push(
-        <CollectionRow
-          id={id}
-          key={id + visibility}
-          visibility={visibility}
-          accessType={accessType}
-        />
-      );
+  const revisions = revisionsEnabled ? findRevisions(collections) : false;
+  sortByCreatedAtDescending(revisions || collections).forEach(
+    ({ id, visibility, revision }) => {
+      if (!displayVisibility || visibility === displayVisibility) {
+        collectionElements.push(
+          <CollectionRow
+            id={id}
+            key={id + visibility}
+            visibility={visibility}
+            accessType={accessType}
+            hasRevision={revision}
+          />
+        );
+      }
     }
-  });
+  );
 
   return collectionElements;
 }
 
 function sortByCreatedAtDescending(
-  collections: CollectionResponse[]
-): CollectionResponse[] {
+  collections: RevisionResponse[]
+): RevisionResponse[] {
   return collections?.sort((a, b) => b.created_at - a.created_at) || [];
 }
 
+function findRevisions(collections: CollectionResponse[]): RevisionResponse[] {
+  const revisionMap = new Map<CollectionResponse, number>();
+  collections.forEach((collection) => {
+    revisionMap.set(collection, revisionMap.get(collection) ?? 1);
```

Comment:
Probably need help understand how this piece of code works in terms of finding whether a collection has a revision or not, since I currently don't have time to test this further 😆 

So far it seems to be looping through `collections` to create `revisionMap`, where key is `collection` and value is always `1`? 

```
// seems like revisionMap.get(collection) can only return `null` or 1?
revisionMap.set(collection, revisionMap.get(collection) ?? 1); 
``` 

And then:

```
// revision will always be false, because number is always 1?
return { ...collection, revision: number > 1 } 
```

Reply:
Total bug, I haven't been able to test it since there's no backend route yet.  I think it should be:
```suggestion
    revisionMap.set(collection, (revisionMap.get(collection) ?? 0) + 1);
```

Reply:
Got it! 

Since the key is a Collection response, which is a pointer to a POJO, it's gonna be unique for every item in `collections`? In other words, what props in a collection object do we use to identify an original and a revision belong to the same collection? Is it `collection.id` or something?

E.g.,

```ts
// do collection and its revision share the same `id`?
const collection = { id: 1, name: 1, ... }
const collectionRevision = { id: 1, name: 'revision', ... }


const collections = [collection, collectionRevision, ... ]

// false, since they are different pointers
console.log(collection === collectionRevision)

// true, since they share the same id, thus if `revisionMap`'s key is `id`, the count will aggregate correctly?
console.log(collection.id === collectionRevision.id) 
```